ksql-cli/src/main/java/io/confluent/ksql/cli/console/cmd/RemoteServerSpecificCommand.java

```diff
@@ -24,18 +24,23 @@
 import io.confluent.ksql.util.ErrorMessageUtil;
 import java.io.PrintWriter;
 import java.util.Objects;
+import java.util.function.Consumer;
 import javax.ws.rs.ProcessingException;
 
 public class RemoteServerSpecificCommand implements CliSpecificCommand {
 
   private final KsqlRestClient restClient;
   private final PrintWriter writer;
+  private final Consumer<Void> resetCliForNewServer;
 
   public RemoteServerSpecificCommand(
       final KsqlRestClient restClient,
-      final PrintWriter writer) {
+      final PrintWriter writer,
+      final Consumer<Void> resetCliForNewServer) {
```

Comment:
We should just add a generic 'Event' interface to our common lib.  Do you fancy doing it? :D